pkgs/by-name/README.md

```diff
@@ -76,6 +76,17 @@ and override its value in [`pkgs/top-level/all-packages.nix`](../top-level/all-p
 }
 ```
 
+### Toplevel argument restrictions
+The list of toplevel arguments should be restricted to packages defined in the
+common nixpkgs package set, and simple `enable*` boolean flags.
```

Comment:
Attributes of type string are employed to indicate options. Emacs expression accepts `toolkit` as "gtk", "motif", "athena"...

And Emacs is outside `by-name`.